### PR TITLE
Add granularity

### PR DESCRIPTION
**Pull request by axsaucedo**
# Adding Granularity

Added granularity support to the /hist/1.0/journey request

### CODE REVIEW

File: common/util.py [hunk 1]
```diff
@@ -9,14 +18,10 @@ def api_bool(value):
     else:
         return False
 
-UID = 1
-RID = 2
-TRAINID = 3
-
-pattern_rid = re.compile("^[0-9]+$")
-pattern_trainid = re.compile("^[0-9][A-Za-z][0-9][0-9]$")
+def validate_service(service):
```
Comment (vladvelici) [inline common/util.py]: good one

File: common/util.py [hunk 2]
```diff
@@ -9,14 +18,10 @@ def api_bool(value):
...
+    return service
 
-def service(s):
+def service_type(s):
```
Comment (vladvelici) [inline common/util.py:24]: Good rename!

File: common/util.py [hunk 3]
```diff
@@ -24,3 +29,14 @@ def service(s):
...
         return TRAINID
     return UID
+
+def validated_granularity(value):
+    GRANUlARITIES = [
+        "day",
+        "week",
+        "month"
+    ]
+    if not value in GRANUlARITIES:
```
Comment (vladvelici) [inline common/util.py:41]: Maybe a value.toLower() might be good to avoid common mistakes.

File: resources/JourneyResource.py [hunk 4]
```diff
@@ -22,23 +28,50 @@
...
+            grouping = "c.tiploc"
+        else:
+            raise Exception("No service or station given")
 
         args = query_parser.parse_args()
 
-        service = args.service
-        station = validate_tiploc(args.station)
+        granularity = validated_granularity(args.granularity)
+        service = validate_service(args.service) if (not service and args.service) else service
```
Comment (vladvelici) [inline resources/JourneyResource.py]: Did you mean `else validate_service(service)`?

(and same on the next line)

Comment (axsaucedo) [inline resources/JourneyResource.py]: ah great point!

File: common/util.py (hunk 3 above)
Comment (axsaucedo) [inline common/util.py:41]: good idea!

**Pull request merged**